adagucserverEC/CDBAdapterPostgreSQL.cpp

```diff
@@ -1080,3 +1084,86 @@ int CDBAdapterPostgreSQL::addFilesToDataBase() {
 #endif
   return 0;
 }
+
+int CDBAdapterPostgreSQL::storeLayerMetadata(const char *datasetName, const char *layerName, const char *metadataKey, const char *metadataBlob) {
+#ifdef MEASURETIME
+  StopWatch_Stop(">CDBAdapterPostgreSQL::storeLayerMetadata");
+#endif
+  CPGSQLDB *dataBaseConnection = getDataBaseConnection();
+  if (dataBaseConnection == NULL) {
+    return -1;
+  }
+
+  CT::string tableColumns("datasetname varchar (255), layername varchar (255), metadatakey varchar (255), updatetime TIMESTAMPTZ, blob JSONB, PRIMARY KEY (datasetname, layername, metadatakey)");
+
+  int status = dataBaseConnection->checkTable("layermetadata", tableColumns.c_str());
+  if (status == 1) {
+    CDBError("\nFAIL: Table layermetadata could not be created: %s", tableColumns.c_str());
+    throw(__LINE__);
+  }
+
+  CT::string updateTime = CTime::currentDateTime().c_str();
+  updateTime.setSize(19);
+  CT::string query;
+  query.print("INSERT INTO layermetadata values (E'%s',E'%s', E'%s', E'%s', E'%s') "
+              "ON CONFLICT (datasetname, layername, metadatakey) "
+              "DO UPDATE SET blob = excluded.blob, updatetime = excluded.updatetime;",
+              datasetName, layerName, metadataKey, updateTime.c_str(), metadataBlob);
+
+  status = dataBaseConnection->query(query.c_str());
+  if (status != 0) {
+    CDBError("Unable to insert records: \"%s\"", query.c_str());
+    throw(__LINE__);
+  }
+#ifdef MEASURETIME
+  StopWatch_Stop("<CDBAdapterPostgreSQL::storeLayerMetadata");
+#endif
+  return 0;
+}
+
+CT::string CDBAdapterPostgreSQL::getLayerMetadata(const char *datasetName, const char *layerName, const char *metadataKey) {
+#ifdef MEASURETIME
+  StopWatch_Stop(">CDBAdapterPostgreSQL::getLayerMetadata");
+#endif
+
+  if (this->layerMetaDataStore == nullptr) {
```

Comment:
This is a cache, right? The idea is to avoid multiple queries to the DB when this function gets called for different layers and `metadataKey`, right?

What if this function get called again with a different `datasetName`?

Reply:
That is true (as all the tests works...), but at the very least the function comments should make this very clear. Or be safe and implement the case by storing the values in a map by`datasetName`.